test/functionalTest/cases/0000_statistics_operation/statistics_with_semaphores.test

```diff
@@ -28,6 +28,12 @@ dbInit CB
 brokerStart CB 0 IPv4 -statSemWait -statTiming
 
 --SHELL--
+echo "+++++ 0. Version - just to make statitstics work also when under valgrind"
+orionCurl --url /version
```

Comment:
You have explained at some moment in the past but I'm unable to remember now... :$

Why GET /version needs to be done to make statistics work under valrgind? (We can talk at skype if it is preferable than using github comments)

Reply:
It's easy.
when we start the broker under valgrind, a GET /version is used to make sure the broker is alive.
That request gets included in the statistics and makes some fields appears under valgrind but not under normal tests. For example, a functest that only uses service path /sp/1 and then asks for metrics:
Under valgrind you will see the "GET /version" (without service path, and the requests of the functest, with /sp/1 as service path.

If I "force" a GET /version inside the functest, then the metrics will still be different, but only in number of requests and sizes. The metrics of "default service path" will be there for both cases and a diff becomes possible

Reply:
Ah, ok... that is I now remember :)

What about of implementing a new operation in harnessFunction.sh (e.g. `valgrindMetricsPrepare`... or a better name) implemented as `curl /version > /dev/null 2>&1`? That would achive the target of prepare the metrics structure as you say and, at the same time, would about noise in the REGEX part of the test.

Reply:
That is one option.
Only one drawback. What if the GET /version fails ...?
No output in the .out file of the test. Could be confusing

Better would be to use `nc` instead of `GET /version` inside 'valgrind-broker-start'.
But, that would give a lot of work fixing functests ...

Or keep as is, at least it is fully controlled.

Not sure which option I prefer. The `nc` option, I think.

Reply:
I also think I prefer `nc` option. But not for this PR, but in a separte issue (to be created).

NTC

Reply:
Let's go for `nc` then.
Lot's of changes in a separate PR ...

Reply:
Issue created at https://github.com/telefonicaid/fiware-orion/issues/3201